tests/integration/cartography/intel/aws/ec2/test_ec2_volumes.py

```diff
@@ -3,123 +3,147 @@
+        ('vol-03', True, None),
+        ('vol-04', True, None),
+        ('vol-09', True, None),
+        ('vol-0df', True, None),
     }
 
-    result = neo4j_session.run(
-        """
-        MATCH (n1:AWSAccount)-[:RESOURCE]->(n2:EBSVolume) RETURN n1.id, n2.id;
-        """,
-    )
-    actual = {
-        (r['n1.id'], r['n2.id']) for r in result
+    # Assert that they are attached to the instances
+    assert check_rels(
+        neo4j_session,
+        'EC2Instance',
+        'instanceid',
+        'EBSVolume',
+        'volumeid',
+        'ATTACHED_TO',
+        rel_direction_right=False,
+    ) == {
+        ('i-01', 'vol-0df'),
+        ('i-02', 'vol-03'),
+        ('i-03', 'vol-09'),
+        ('i-04', 'vol-04'),
     }
 
-    assert actual == expected
-
-
-@patch.object(cartography.intel.aws.ec2.instances, 'get_ec2_instances', return_value=DESCRIBE_INSTANCES['Reservations'])
-def test_load_volume_to_instance_rels(mock_get_instances, neo4j_session):
-    # Arrange: Load in ec2 instances first
-    boto3_session = MagicMock()
-    sync_ec2_instances(
+    # Act
+    sync_ebs_volumes(
         neo4j_session,
         boto3_session,
         [TEST_REGION],
         TEST_ACCOUNT_ID,
         TEST_UPDATE_TAG,
         {'UPDATE_TAG': TEST_UPDATE_TAG, 'AWS_ID': TEST_ACCOUNT_ID},
     )
-    # Prep the volume data
-    raw_volumes = tests.data.aws.ec2.volumes.DESCRIBE_VOLUMES
-    transformed_volumes = cartography.intel.aws.ec2.volumes.transform_volumes(raw_volumes, TEST_REGION, TEST_ACCOUNT_ID)
 
-    # Act
-    cartography.intel.aws.ec2.volumes.load_volume_relationships(
-        neo4j_session,
-        transformed_volumes,
-        TEST_UPDATE_TAG,
-    )
+    # Assert that additional fields such as `encrypted` have been added by sync_ebs_volumes(), while
+    # deleteontermination has not been overwritten with None by sync_ebs_volumes()
+    assert check_nodes(neo4j_session, 'EBSVolume', ['id', 'deleteontermination', 'encrypted']) == {
+        # Attached to the instances initially
+        ('vol-04', True, None),
+        ('vol-09', True, None),
+        # Added by ebs sync
+        ('vol-03', True, True),
+        ('vol-0df', True, True),
+    }
 
-    # Assert
-    result = neo4j_session.run(
-        """
-        MATCH (n1:EC2Instance)<-[:ATTACHED_TO_EC2_INSTANCE]-(n2:EBSVolume) RETURN n1.id, n2.id;
-        """,
-    )
-    expected = {
-        ('i-01', 'v-01'),
-        ('i-02', 'v-02'),
+    # Assert that they are still attached to the instances
+    assert check_rels(
+        neo4j_session,
+        'EC2Instance',
+        'instanceid',
+        'EBSVolume',
+        'volumeid',
+        'ATTACHED_TO',
+        rel_direction_right=False,
+    ) == {
+        ('i-01', 'vol-0df'),
+        ('i-02', 'vol-03'),
+        ('i-03', 'vol-09'),
+        ('i-04', 'vol-04'),
     }
-    actual = {
-        (r['n1.id'], r['n2.id']) for r in result
+
+    # Assert that the account to volume rels exist
+    assert check_rels(
```

Comment:
We should maybe do this check after syncing instances, but before syncing the volumes